tests/end_to_end/test_suites/wf_local_func_tests.py

```diff
@@ -0,0 +1,249 @@
+        flflow.run()
+    log.info("Successfully ended test_reference_exclude_flow")
+
+
+@pytest.mark.parametrize("fx_local_federated_workflow", [("init_collaborator_private_attr_name", "str", None )], indirect=True)
+def test_subset_collaborators(request, fx_local_federated_workflow):
+    """
+    Test the subset of collaborators in a federated workflow.
+
+    Parameters:
+        request (FixtureRequest): The request fixture provides information about the requesting test function.
+        fx_local_federated_workflow (Fixture): The fixture for the local federated workflow.
+
+    Tests:
+        - Ensure the test starts and ends correctly.
+        - Verify the number of collaborators matches the expected subset.
+        - Check that the flow runs for each subset collaborator.
+    """
+    log.info("Starting test_subset_collaborators")
+    collaborators = fx_local_federated_workflow.collaborators
+
+    random_ints = random.sample(range(1, len(collaborators) + 1), len(collaborators))
+
+    collaborators = fx_local_federated_workflow.runtime.collaborators
+    for round_num in range(len(collaborators)):
+        log.info(f"Starting round {round_num}...")
+
+        if os.path.exists(".metaflow"):
+            shutil.rmtree(".metaflow")
+
+        flflow = TestFlowSubsetCollaborators(checkpoint=True, random_ints=random_ints)
+        flflow.runtime = fx_local_federated_workflow.runtime
+        flflow.run()
+        subset_collaborators = flflow.subset_collaborators
+        collaborators_ran = flflow.collaborators_ran
+        random_ints = flflow.random_ints
+        random_ints.remove(len(subset_collaborators))
+
+        step = Step(
+            f"TestFlowSubsetCollaborators/{flflow._run_id}/"
+            + "test_valid_collaborators"
+        )
+
+        assert len(list(step)) == len(subset_collaborators), (
+                f"...Flow only ran for {len(list(step))} "
+                + f"instead of the {len(subset_collaborators)} expected "
+                + f"collaborators- Testcase Failed."
+            )
+        log.info(
+            f"Found {len(list(step))} tasks for each of the "
+            + f"{len(subset_collaborators)} collaborators"
+        )
+        log.info(f'subset_collaborators = {subset_collaborators}')
+        log.info(f'collaborators_ran = {collaborators_ran}')
+        for collaborator_name in subset_collaborators:
+            assert collaborator_name in collaborators_ran, (
+                f"...Flow did not execute for "
+                + f"collaborator {collaborator_name}"
+                + f" - Testcase Failed."
+            )
+
+    log.info(
+        f"Testing FederatedFlow - Ending test for validating "
+        + f"the subset of collaborators.")
+    log.info("Successfully ended test_subset_collaborators")
+
+
+def test_private_attr_wo_callable(request, fx_local_federated_workflow_prvt_attr):
+    """
+    Set private attribute without callable function i.e thru direct assignment
```

Comment:
The word 'thru' is informal, so let's use the word 'through' wherever applicable.